libr/bin/pdb/pdb.c

```diff
@@ -1172,7 +1172,7 @@ static void print_gvars(R_PDB *pdb, ut64 img_base, int format) {
 }
 
 ///////////////////////////////////////////////////////////////////////////////
-R_API bool init_pdb_parser(R_PDB *pdb, const char *filename) {
+R_API bool init_pdb_parser_with_buf(R_PDB *pdb, RBuffer* buf) {
```

Comment:
all the public pdb apis should start with pdb_ not init_pdb.

Actually this should be a constructor ,right? In my head i can thnk in something like:

```cs
var p = new PDB();
p.load_from_buffer(buf);
p.load_from_file(fileName); // w can user r_file_slurp_buffer() instead of having two constructors
```

Reply:
having multiple constructors is a problem because we need to fill some pointers (cb_printf... anything else?)

im fine on having this merged, but i do think the PDB api should be redesigned to be reusable from the outside

Reply:
Yeah, I agree. Even with this PR it's only barely usable: I still need to include `pdb_types.h` and most of `print_gvars` to get useful info out of it.